### PR TITLE
Adds Incomplete Request Count Tracking

### PR DESCRIPTION
**Pull request by nija123098**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature thoroughly
  * http://imgur.com/gallery/mA8hB
  * http://imgur.com/gallery/O8EY4

**Issues Fixed:** None

### Changes Proposed in this Pull Request
* adds RequestBuffer.incompleteRequestCount()

### CODE REVIEW

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 1]
```diff
@@ -51,11 +53,21 @@ public static void request(IVoidRequest request) {
 	}
 
 	/**
+	 * This returns the number of incomplete requests.
+	 *
+	 * @return The number of incomplete requests.
+	 */
+	public static int incompleteRequestCount(){
```
Comment (chrislo27) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: Space missing between ) and {.

Comment (chrislo27) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: Shouldn't this be something like `getIncompleteRequestsCount`?

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 2]
```diff
@@ -125,8 +137,10 @@ public RequestFuture(IRequest<T> request) {
 		 */
 		@Override
 		public boolean cancel(boolean mayInterruptIfRunning) {
-			if (!isDone())
+			if (!isDone()){
```
Comment (chrislo27) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: Missing space between ) and {.

**Comment by chrislo27** [comment]
I will also note that we are considering using ScheduledExecutorService instead of Timer as a rewrite, so this will not be merged until we get Austin's word.

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 3]
```diff
@@ -51,11 +53,21 @@ public static void request(IVoidRequest request) {
 	}
 
 	/**
+	 * This returns the number of incomplete requests.
+	 *
+	 * @return The number of incomplete requests.
+	 */
+	public static int getIncompleteRequestCount() {
+		return incompleteRequestCount;
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: You could also just count the entries in the `requests` field. Much simpler and would be more maintainable in the future.

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 4]
```diff
@@ -51,11 +53,21 @@ public static void request(IVoidRequest request) {
 	}
 
 	/**
+	 * This returns the number of incomplete requests.
+	 *
+	 * @return The number of incomplete requests.
+	 */
+	public static int getIncompleteRequestCount() {
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: You could also just count the entries in the `requests` field. Much simpler and would be more maintainable in the future.

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 5]
```diff
@@ -58,7 +56,7 @@ public static void request(IVoidRequest request) {
 	 * @return The number of incomplete requests.
 	 */
 	public static int getIncompleteRequestCount() {
-		return incompleteRequestCount;
+		return requests.size();
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: This is not enough. That gets the size of the bucket-requests map. But there could be more than one request queued per bucket.

Comment (nija123098) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: I was so confused why you might cancel the requests if more than one had failed, then I woke up to the remembrance that the keys were mapped to lists.

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 6]
```diff
@@ -56,7 +56,11 @@ public static void request(IVoidRequest request) {
 	 * @return The number of incomplete requests.
 	 */
 	public static int getIncompleteRequestCount() {
-		return requests.size();
+		final int[] count = {0};
```
Comment (austinv11) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: You should use an `AtomicInteger` as this is the use case for such a class as well as it being more consistent with the rest of the source.

**Pull request merged**